SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp

```diff
@@ -339,6 +339,32 @@ bool FSpatialGDKEditorToolbarModule::ValidateGeneratedLaunchConfig() const
 	const USpatialGDKSettings* SpatialGDKRuntimeSettings = GetDefault<USpatialGDKSettings>();
 	const FSpatialLaunchConfigDescription& LaunchConfigDescription = SpatialGDKEditorSettings->LaunchConfigDesc;
 
+	// Ensure that there is an Unreal Server worker in the launch config worker list.
+	if (!LaunchConfigDescription.Workers.ContainsByPredicate([](const FWorkerTypeLaunchSection& Worker) { return Worker.WorkerTypeName == SpatialConstants::ServerWorkerType; }))
+	{
+		const EAppReturnType::Type Result = FMessageDialog::Open(EAppMsgType::YesNo, FText::FromString(FString::Printf(TEXT("The autogenerated launch config does not have a worker section for the UnrealWorker. This is required to launch a deployment. \n\nDo you want to configure your launch config settings now?"))));
+
+		if (Result == EAppReturnType::Yes)
+		{
+			FModuleManager::LoadModuleChecked<ISettingsModule>("Settings").ShowViewer("Project", "SpatialGDKEditor", "Editor Settings");
+		}
+
+		return false;
+	}
+
+	// Ensure that there are no additional server workers in the launch config.
+	if(!SpatialGDKRuntimeSettings->bUsingOffloading && LaunchConfigDescription.Workers.Num() > 1)
+	{
+		const EAppReturnType::Type Result = FMessageDialog::Open(EAppMsgType::YesNo, FText::FromString(FString::Printf(TEXT("Offloading is currently disabled but the autogenerated lauch config contains worker sections for offloaded workers. Either enable offloading or remove the worker sections for the offloaded workers.\n\nDo you want to configure your launch config settings now?"))));
```

Comment:
IMO it's fine to have the config description for the workers. Turning off offloading should prevent only the launching of these workers (e.g. from the "workers to launch" part), but not definitions of them.

I guess you can check if any are configured to be "managed launched by deployment start", though.

Reply:
The reason why I avoided putting in that check in this PR is beacuse it would require and engine change, I think this is something that we should definitely do when we productize

Reply:
By "that check" do you mean "workers to launch"? If that is the case then that further eliminated the need for this check, because if you don't check that one but check this one you'd be trying to launch workers that are not declared within the launch config descriptors which is a worse error